web/frontend/src/components/audio-input/index.svelte

```diff
@@ -29,52 +31,42 @@ const handleTrack = (event: unknown) => {
   audio.srcObject = eventStream;
 };
 
-const toggleExpand = async () => {
-  isOn = !isOn;
+const handleToggle = (event: CustomEvent<{ open: boolean }>) => {
+  expanded = event.detail.open;
+};
 
-  streamClient.on('track', handleTrack);
+useConnect(() => {
+  connected = true;
 
-  if (isOn) {
-    try {
-      await streamClient.add(name);
-    } catch (error) {
-      displayError(error as ServiceError);
-    }
-    return;
+  return () => {
+    connected = false;
   }
+});
 
-  try {
-    await streamClient.remove(name);
-  } catch (error) {
-    displayError(error as ServiceError);
-  }
-};
 
-useDisconnect(() => {
-  streamClient.off('track', handleTrack);
+onMount(() => {
+  streamClient.on('track', handleTrack);
+  return () => streamClient.off('track', handleTrack)
 });
 
+$: if (connected && expanded && !added) {
+  streamClient.add(name).catch((error) => displayError(error as ServiceError));
+  added = true;
+} else if (added) {
+  streamClient.remove(name).catch((error) => displayError(error as ServiceError));
+  added = false;
+}
+
 </script>
 
-<Collapse title={name}>
+<Collapse title={name} on:toggle={handleToggle}>
   <v-breadcrumbs slot="title" crumbs="audio_input" />
   <div class="h-auto border border-t-0 border-medium p-2">
-    <div class="flex items-center gap-2">
-      <v-switch
```

Comment:
This switch was pointless, pressing the play / pause button achieves the same goal.